src/components/HistoricalForecast/HistoricalForecast.tsx

```diff
@@ -1,130 +1,48 @@
+          variant: 'error',
+        });
+      } finally {
+        setIsLoading(false);
+      }
+    };
+    fetchForecast();
+  }, [selectedSearchParams, enqueueSnackbar]);
 
   return (
-    <Formik<FormValuesType>
-      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-      // @ts-ignore
-      // We ignore ts rule, because initial values should be optional, but in Formik they must exactly match the validation scheme
-      initialValues={INITIAL_FORM_VALUES}
-      onSubmit={handleSubmit}
-      validationSchema={VALIDATION_SCHEMA}>
-      {({ isSubmitting }) => (
-        <Form>
-          <Grid
-            container
-            spacing={2}>
-            <Grid
-              item
-              xs={12}>
-              <Field
-                component={SelectField<LanguageOption>}
-                label={LANGUAGE_CHOICE_LABEL}
-                labelId={LANGUAGE_CHOICE_LABEL_ID}
-                name='language'
-                options={LANGUAGE_OPTIONS}
-              />
-            </Grid>
-            <Grid
-              item
-              xs={2.5}>
-              <Field
-                disableHighlightToday
-                css={classes.startDateField}
-                component={DateField}
-                label='Start'
-                maxDate={yesterday}
-                name='startDate'
-                type='date'
-              />
-            </Grid>
-            <Grid
-              item
-              xs={2.5}>
-              <Field
-                disableHighlightToday
-                css={classes.endDateField}
-                component={DateField}
-                label='End'
-                maxDate={yesterday}
-                name='endDate'
-                type='date'
-              />
-            </Grid>
-            <Grid
-              item
-              xs={12}>
-              <Field
-                css={classes.location}
-                component={LocationAutoCompleteField}
-                id={LOCATION_AUTOCOMPLETE}
-                name='location'
-              />
-            </Grid>
-            <Grid
-              item
-              xs={12}>
-              <Field
-                component={SelectField<TemperatureUnitOption>}
-                labelId={TEMPERATURE_UNITS_LABEL_ID}
-                label={TEMPERATURE_UNITS_LABEL}
-                name='temperatureUnit'
-                options={TEMPERATURE_UNITS_OPTIONS}
-              />
-            </Grid>
-            <Grid item>
-              <LoadingButton
-                loading={isSubmitting}
-                type='submit'
-                variant='contained'>
-                Submit
-              </LoadingButton>
-            </Grid>
-          </Grid>
-        </Form>
+    <>
+      <HistoricalWeatherForm setSearchParams={onSelectSearchParams} />
+      {forecast && selectedSearchParams && (
+        <HistoricalWeatherWidget
+          forecast={forecast}
+          loadingStatus={isLoading}
```

Comment:
`isLoading`. Good practice to append to a name of boolean props `is` or `has` in order to hint their type